setup.py

```diff
@@ -146,6 +152,11 @@ def run(self):
         target = {"min": "build", "dev": "build:dev"}
 
         try:
+            install_google_fonts(
+                files(resources),
+                files(resources),
+                DEFAULT_THEME_OPTIONS["font"].values(),
+            )
```

Comment:
I don't think this will really solve the problem reported in #222. If a CI installs this theme, then the install process will error out if the CI runner doesn't have access to Google's font API (as similarly reported in #222).

I was under the impression that we'd have to include copies of the default fonts with the theme src to properly address #222. I do appreciate the attempt to update the default fonts at pkg install time, but it isn't really feasible (with respect to #222).

#222 would be better resolved by updating the default fonts' cache when `npm run build` is performed.

Reply:
When installing from the git repository directly, we have to download the npm packages (which requires internet access).  However, the wheel package that we publish to PyPI includes the pre-built javascript and css bundles, and icons, and I think the idea here is that it could also include the Google fonts.  So it would in fact solve the problem reported in #222 as long as the user does not wish to change the fonts from their default values.

Reply:
Yeah that was my thinking exactly.

I'm not sure if installing from git is preferred when the users' env has limited access to other domains.

Reply:
Installing from git/zip is also common for envs that don't have python's pip tool. For example, the python executable shipped with Ubuntu usually doesn't have `pip` or `venv` and the admin group (needed to use `apt`) isn't viable for the active user(s). Sure, there are alternative install methods (docker containers), but it usually falls on pkg maintainers when all else fails.

Reply:
@2bndy5 I might be wrong but my understanding is that users who don't have pip or venv and who are not in the admin group will need to install npm and the JS deps in order to install this package. Therefore, they have more things to worry about than whether their env/runner has internet access to Google. Setting that up falls into "packaging", which can be done by the distro maintainers or by the end user.

Hence, that's a different use case than the one covered in this PR. I believe that this PR should be compatible with that one, but it should not be constrained by envs not having access to Google at build time.

Reply:
> I believe that this PR should be compatible with that one, but it should not be constrained by envs not having access to Google at build time.

That is the exact problem that #222 poses. The OP of this PR aims to implement a solution discussed in #222. I feel like we're getting off track again (you and I seem to have a "chemistry" 😜 ).

Reply:
BTW, if installing from a zip, the archive should be in the form of the sdist deployed to pypi which shouldn't need npm or the repo's JS deps because `npm run build` bundles all of that stuff into the sphinx_immaterial directory _before_ compressing the archive. Then the setup.py just indicates (via [`package_data`](https://setuptools.pypa.io/en/latest/userguide/datafiles.html#package-data)) what is contained in the sphinx_immaterial pkg.

Reply:
Let's see if I'm getting it right:

- #222 is the discussion of a problem and 5 different solutions are proposed, each covering slighly different use cases.
- #232, this PR, implements one of the 5 solutions to support one of the use cases.
- We want to make sure that implementing this use case does not break all other 4 use cases. We all agree on that.

What's the need to portrait this as a competitive solution instead of a collaborative one? I'm referring to *"https://github.com/jbms/sphinx-immaterial/issues/222 would be **better** resolved by updating the default fonts' cache when npm run build is performed"* (bolds added by me). Precisely, as discussed in #222, it is only solved by covering the five use cases, or by making it explicit which of them are not to be supported (and why). Implementing a single one and ignoring others does not resolve #222, no matter which one it is. This PR is "related to #222", not closing or fixing it.

> I feel like we're getting off track again (you and I seem to have a "chemistry" 😜 ).

It's ok, it's just a matter of experience in the open-source galaxy 😃. We have very different points of view about the project. You try to bring attention towards other related maintenance/development issues which you feel important, and I'm trying to ponderate attention from an external position. In several other projects, I do play the role you are playing here.

Reply:
> BTW, if installing from a zip, the archive should be in the form of the sdist deployed to pypi which shouldn't need npm or the repo's JS deps because npm run build bundles all of that stuff into the sphinx_immaterial directory before compressing the archive. Then the setup.py just indicates (via [package_data](https://setuptools.pypa.io/en/latest/userguide/datafiles.html#package-data)) what is contained in the sphinx_immaterial pkg.

This is interesting! So, the proposal is to install an sdist instead of the zip of this repo? Where would that sdist zip be hosted?

Reply:
Oh, they aren't 5 use cases; they're [5 possible solutions](https://github.com/jbms/sphinx-immaterial/issues/222#issuecomment-1446913574) to the reported problem. We agreed to pursue solution 5 because it is the most unobtrusive approach. The other solutions (except solution 1) would still be supported, but they require user-implementation. Solution 1 has been unsupported ever since the external resources ext was added. Consequently, not supporting solution 1 has instigated #222...

I'm not sure how other spoken languages work out, but English is often too imprecise for us analytical thinkers 🤓 

---
FYI, we had discussed something like solution 5 when the external resources ext was added (#144), but because that was lumped into the theme's custom graphviz addition, we decided it deserved its own PR.

Reply:
Agree. Code speaks better. 

Is the npm based approach something you expect to do in the following weeks? We (I) can hold this discussion until we can better see and understand what's the difference.

Reply:
I have an idea about using the fontsource npm dists for [Robot](https://www.npmjs.com/package/@fontsource/roboto) and [Robot Mono](https://www.npmjs.com/package/@fontsource/roboto-mono). But I haven't tackled how to integrate it into google_fonts.py yet.

Reply:
> With "in the form" you mean "something that looks like it" or "it will be uploaded to pypi and installable through pip"?

I mean the archive would be the actual sdist that is uploaded to pypi but agnostic to the compression used (tar.gz vs zip).

Reply:
I completely missed that comment, so I'm answering it in parts...

> Where would that sdist zip be hosted?

Currently, we don't publish releases on github, but we could upload the archive as a release asset if we do. Maybe that is something that should be discussed as part of #223. However, installing from git does require npm and stuff.

Reply:
@2bndy5 any update to this feature? Is there an alternative implementation?

Reply:
If you or @glatosinski could satisfy the review concerns here, then I'd be happy to move this PR forward and implement the switch to fontsource CDN in a separate PR.

I'm currently enthralled with exploring my idea for #261.

Reply:
Ok, I'll look into it